### ISSUE TITLE
🐛 logger middleware doesn't work well with etag

### ISSUE THREAD

**Issue by dhcmrlchtdj** [gofiber/fiber#909]
**Fiber version**
v2.0.6

**Issue description**
`curl -v http://localhost:3000 -H 'If-None-Match: "9-450763418"'` return `304 Not Modified` with empty body.
but logger print `GET / | status: 200 | bytesSent: 9`, the `status` and `bytesSent` are wrong.

**Code snippet**

```go
package main

import (
	"github.com/gofiber/fiber/v2"
	"github.com/gofiber/fiber/v2/middleware/logger"
)

func main() {
	app := fiber.New(fiber.Config{ETag: true})

	app.Use(logger.New(logger.Config{
		Format: "${method} ${path} | status: ${status} | bytesSent: ${bytesSent}\n",
	}))
	app.Get("/", func(c *fiber.Ctx) error {
		return c.SendString("with etag")
	})

	app.Listen(":3000")
}
```

**Comment by dhcmrlchtdj** [comment]
a possible sulotion
export etag functionality as a middleware, which can be mounted after logger middleware.
for compatibility, we can mount etag middleware inside `app.Listen`, and add a deprecate warning to etag option.

**Comment by kiyonlin** [comment]
> a possible sulotion
> 
> export etag functionality as a middleware, which can be mounted after logger middleware.
> 
> for compatibility, we can mount etag middleware inside `app.Listen`, and add a deprecate warning to etag option.

@dhcmrlchtdj Maybe we can use etag option to determine whether we should hit etag middleware

**Comment by dhcmrlchtdj** [comment]
> for compatibility, we can mount etag middleware inside `app.Listen`

hmm,  it is infeasible.

```
package github.com/gofiber/fiber/v2
	imports github.com/gofiber/fiber/v2/middleware/etag
	imports github.com/gofiber/fiber/v2: import cycle not allowed
```

**Comment by kiyonlin** [comment]
`etag` middleware should be placed in fiber core folder.

**Comment by Fenny** [comment]
Thanks @dhcmrlchtdj for this bug report, the idea to add the `ETag` logic as a middleware to provide more flexibility was spot on.
https://github.com/gofiber/fiber/pull/926 will be tagged in version `v2.1.0` and should solve this problem, I will leave this issue open until it's released 👍

**Comment by Fenny** [comment]
[ETag middleware](https://github.com/gofiber/fiber/tree/master/middleware/etag) is now available in [v2.1.0](https://github.com/gofiber/fiber/releases/tag/v2.1.0), you can place this mw after the logger 👍
